File: src/C#/BehavioralPatterns/Mediator/MultipleExamples/CommunicationHub/Mediators/Group.cs
using MultipleExamples.CommunicationHub.Participants;
using System;
using System.Collections.Generic;
using System.Text;

namespace MultipleExamples.CommunicationHub.Mediators
{
    public class Group
    {
        public List<Participant> Participants { get; set; } = new List<Participant>();

        public bool DoesParticipantExist(Participant participant) => Participants.Contains(participant);
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/CommunicationHub/mediators/group.js

class Group {
    constructor() {
        this.participants = [];
    }

    get participants() {
        return this._participants;
    }

    set participants(value) {
        this._participants = value;
    }

    doesParticipantExist(participant) {
        return this.participants.includes(participant);
    }
}

module.exports = Group;
